### PR TITLE
Fix issues with PL 1.8

### PR DESCRIPTION
**Pull request by SeanNaren**
# What does this PR do ?

Attempts to fix all issues after many deprecations/lightning lite becoming the backbone for a lot of the underlying pytorch lightning code.

cc @titu1994 

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -658,24 +657,6 @@ def get_git_diff():
         return "{}\n".format(err.output.decode("utf-8"))
 
 
-class LoggerList(_LoggerCollection):
```
Comment (SeanNaren) [inline nemo/utils/exp_manager.py:661]: This is not available in Lightning any more (it was moved to the `_grave_yard` which is pretty aggressive). I've removed it for now, but I need to check what the disadvantage of this is.

File: nemo/collections/tts/models/fastpitch.py [hunk 2]
```diff
@@ -228,11 +228,10 @@ def tb_logger(self):
...
                 return None
             tb_logger = self.logger.experiment
-            if isinstance(self.logger, LoggerCollection):
-                for logger in self.logger:
-                    if isinstance(logger, TensorBoardLogger):
-                        tb_logger = logger.experiment
-                        break
+            for logger in self.trainer.loggers:
+                if isinstance(logger, TensorBoardLogger):
+                    tb_logger = logger.experiment
```
Comment (titu1994) [inline nemo/collections/tts/models/fastpitch.py:233]: I think they deprecated LoggerCollection - should we upgrade and stop using it too ? Maybe just pass list of callback to trainer ?

File: nemo/core/config/hydra_runner.py [hunk 3]
```diff
@@ -67,6 +67,9 @@ def wrapper(cfg_passthrough: Optional[DictConfig] = None) -> Any:
                 # Set run.dir ONLY for ExpManager "compatibility" - to be removed.
                 overrides.append("hydra.run.dir=.")
 
+                # Set working directory to the job's output directory
+                overrides.append("hydra.job.chdir=True")
```
Comment (titu1994) [inline nemo/core/config/hydra_runner.py:71]: Wait why do we want this ? We Don't want to chdir

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (titu1994) [reply nemo/utils/exp_manager.py:661]: That's painful, though I wonder what we needed Nemo name and version for

**Comment by SeanNaren** [comment]
In this PR: https://github.com/Lightning-AI/lightning/pull/11617

Lightning changed behaviour on how we start the other sub-processes with Hydra. This conflicts with how our hydra runner operates, need to investigate further on a solution cc @titu1994

**Comment by SeanNaren** [comment]
This PR should hopefully unblock the move to 1.8: https://github.com/Lightning-AI/lightning/pull/15737#pullrequestreview-1188782551

We'll just wait till it gets into a minor release version (which hopefully would be tomorrow).

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (SeanNaren) [reply nemo/utils/exp_manager.py:694]: From my look around, they were not being used

File: Jenkinsfile [hunk 4]
```diff
@@ -225,9 +225,11 @@ pipeline {
         stage('Speech to Text EMA') {
           steps {
             sh 'python examples/asr/asr_ctc/speech_to_text_ctc.py \
+            --config-path="../conf/" --config-name="config" \
```
Comment (SeanNaren) [inline Jenkinsfile:228]:
```suggestion
```

File: nemo/core/classes/exportable.py [hunk 5]
```diff
@@ -128,7 +129,7 @@ def _export(
             # Set module mode
             with torch.onnx.select_model_mode_for_export(
                 self, training
-            ), torch.inference_mode(), torch.no_grad(), torch.jit.optimized_execution(True):
+            ), torch.inference_mode(), torch.no_grad(), torch.jit.optimized_execution(True), _jit_is_scripting():
```
Comment (titu1994) [inline nemo/core/classes/exportable.py:132]: Hmm we don't use lightning during export, is this method safe ?

File: tests/hydra/test_hydra_runner.py [hunk 6]
```diff
@@ -41,8 +41,6 @@ def test_config1(self):
         # Run the call as subprocess.
         subprocess.check_call(call, shell=True, stdout=sys.stdout, stderr=sys.stdout)
 
-        # Make sure that .hydra dir is not present.
-        assert not path.exists(f".hydra")
```
Comment (titu1994) [inline tests/hydra/test_hydra_runner.py:45]: Why is assert being removed ?

**Review by titu1994 — COMMENTED** [review]
Overall, incredible work @SeanNaren ! Minor comments and needs one other reviewer since it's a core change

File: tests/hydra/test_hydra_runner.py (hunk 6 above)
Comment (SeanNaren) [reply tests/hydra/test_hydra_runner.py]: Ah forgot to revert, have done thanks!

File: nemo/core/classes/exportable.py (hunk 5 above)
Comment (SeanNaren) [reply nemo/core/classes/exportable.py:131]: Would the tests have confirmed if this is fine? The underlying code is simple to what is happening:

In Lightning:
```python
@contextmanager
def _jit_is_scripting() -> Generator:
    """Workaround for https://github.com/pytorch/pytorch/issues/67146."""
    LightningModule._jit_is_scripting = True
    try:
        yield
    finally:
        LightningModule._jit_is_scripting = False
```

Comment (titu1994) [reply nemo/core/classes/exportable.py:131]: Hmm i dont think it would be caught in tests, lets keep it for now

**Pull request merged**